File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:

        stack_of_temps = []

        result = [0] * len(temperatures)
        
        for index, temp in enumerate(temperatures):
            while (stack_of_temps and temp > stack_of_temps[-1][1]): 
                stackIndex, stackTemp = stack_of_temps.pop()
                result[stackIndex] = index - stackIndex
                
            stack_of_temps.append((index, temp))
        return result

File: Data Structures & Algorithms/concatenation-of-array/submission-0.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        
        int N = nums.size();
        std::vector<int> result(N * 2, 0);

        for (int i{}; i < N; i++){
            result[i] = nums[i];
            result[i + N] = nums[i];
        }

        return result;
    }
};